presto-main/src/main/java/com/facebook/presto/sql/planner/plan/SpatialJoinNode.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.plan;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.tree.Expression;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class SpatialJoinNode
+        extends PlanNode
+{
+    private final JoinNode.Type type;
+    private final PlanNode left;
+    private final PlanNode right;
+    /**
+     * List of output symbols produced by join. Output symbols
+     * must be from either left or right side of join. Symbols
+     * from left join side must precede symbols from right side
+     * of join.
+     */
```

Comment:
@kokosing @sopel39 I copy-pasted this from `JoinNode`. I'm seeing that Karol added this to `JoinNode` in https://github.com/mbasmanova/presto/commit/2e9e968df23e6b9b1576e0fee9883c751231bc70 Karol, could you comment on this? Why do we need this in `JoinNode`?

Reply:
This was recently removed from `JoinNode`, remove it from here also.

Reply:
I aggree. I made the same mistake when extending `TableWriterNode` for statstics. It is error prone. Because in case of symbols transformation you need to keep in mind to update this list as well.

Reply:
@kokosing Grzegorz, I'm happy to remove this, but I don't see it removed from `JoinNode`. Is this change in the master?

Reply:
The left comment in JoinNode is obsolete, it should be removed with https://github.com/prestodb/presto/commit/ea3e61ecbe970bbf248efc487cedd88db567da4f

Reply:
@kokosing Thanks for the pointer and thanks for removing this comment from the `JoinNode`. I removed it from `SpatialJoinNode` as well.